example/lib/ornaments.dart

```diff
@@ -0,0 +1,285 @@
+  OrnamentPosition scaleBarPosition = OrnamentPosition.TOP_LEFT;
+  OrnamentPosition logoPosition = OrnamentPosition.BOTTOM_LEFT;
+  OrnamentPosition attributionPosition = OrnamentPosition.BOTTOM_LEFT;
+
+  _onMapCreated(MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+
+    mapboxMap.setCamera(CameraOptions(
+      center: Point(
+          coordinates: Position(
+        -0.11968,
+        51.50325,
+      )).toJson(),
+      padding: null,
+      anchor: null,
+      zoom: 12,
+      bearing: 40,
+      pitch: null,
+    ));
+
+    mapboxMap.compass.updateSettings(CompassSettings(
+      position: compassPosition,
+      enabled: showOrnaments,
+      marginBottom: 10,
+      marginLeft: 10,
+      marginTop: 10,
+      marginRight: 10,
+    ));
+
+    mapboxMap.scaleBar.updateSettings(ScaleBarSettings(
+      position: scaleBarPosition,
+      enabled: showScaleBar,
+      marginBottom: 20,
+      marginLeft: 20,
+      marginTop: 20,
+      marginRight: 20,
+    ));
+
+    mapboxMap.logo.updateSettings(LogoSettings(
+      position: logoPosition,
+      marginBottom: 30,
+      marginLeft: 30,
+      marginTop: 30,
+      marginRight: 30,
+    ));
+
+    mapboxMap.attribution
+        .updateSettings(AttributionSettings(position: attributionPosition));
+  }
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  Widget _getCompassSettings() {
+    return TextButton(
+      child: Text('get compass settings'),
+      onPressed: () {
+        mapboxMap?.compass.getSettings().then(
+            (value) => ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+                  content: Text("""
+                  Ornaments settings : 
+                    clickable : ${value.clickable}, 
+                    enabled : ${value.enabled}
+                    marginBottom : ${value.marginBottom}
+                    marginLeft : ${value.marginLeft}
+                    marginRight : ${value.marginRight}
+                    marginTop : ${value.marginTop}
+                    opacity : ${value.opacity},
+                    position :  ${value.position}
+                    rotation : ${value.rotation}
+                    """
+                      .trim()),
+                  backgroundColor: Theme.of(context).primaryColor,
+                  duration: Duration(seconds: 2),
+                )));
+      },
+    );
+  }
+
+  _toggleOrnaments() {
+    return TextButton(
+      child: Text('toggle compass'),
+      onPressed: () {
+        setState(() {
+          if (showOrnaments) {
+            showOrnaments = false;
+          } else {
+            showOrnaments = true;
+          }
+        });
+      },
+    );
+  }
+
+  Widget _moveOrnaments() {
+    return TextButton(
+      child: Text('move compass'),
+      onPressed: () {
+        setState(() {
+          switch (compassPosition) {
+            case OrnamentPosition.BOTTOM_LEFT:
```

Comment:
nit:
make some map of `current position` -> `new position` and set new position in one line

Reply:
IMO not worth it, enum isn't that long and switch is neither, lets keep it